webapp/src/main/java/io/github/microcks/util/SoapMessageValidator.java

```diff
@@ -41,166 +41,184 @@
-         throws XmlException {
-      //
-      WsdlContext ctx = new WsdlContext(wsdlUrl);
-      List<XmlError> errors = new ArrayList<XmlError>();
-      ctx.getSoapVersion().validateSoapEnvelope(message, errors);
-      
-      log.debug("SoapEnvelope validation errors: " + errors.size());
-   
-      if (validateMessageBody){
-         // Create XmlBeans object for the soap message.
-         XmlOptions xmlOptions = new XmlOptions();
-         xmlOptions.setLoadLineNumbers();
-         xmlOptions.setLoadLineNumbers( XmlOptions.LOAD_LINE_NUMBERS_END_ELEMENT );
-         XmlObject xml = XmlUtils.createXmlObject(message, xmlOptions);
-
-         // Build the QName string of the part name. Ex: {http://www.github.com/lbroudoux/service}sayHello
-         String fullPartName = "{" + partNamespace + "}" + partName;
-         
-         // Extract the corresponding part from soap body.
-         XmlObject[] paths = xml.selectPath( "declare namespace env='" + ctx.getSoapVersion().getEnvelopeNamespace() + "';" 
-               + "declare namespace ns='" + partNamespace + "';" + "$this/env:Envelope/env:Body/ns:" + partName);
-            
-         SchemaGlobalElement elm;
-         try {
-            elm = ctx.getSchemaTypeLoader().findElement(QName.valueOf(fullPartName));
-         } catch (Exception e) {
-            log.error("Exception while loading schema information for " + fullPartName, e);
-            throw new XmlException("Exception while loading schema information for " + fullPartName, e);
-         }
-         
-         if ( elm != null ){
-            validateMessageBody(ctx, errors, elm.getType(), paths[0]);
-      
-            // Ensure no other elements in body.
-            NodeList children = XmlUtils.getChildElements((Element) paths[0].getDomNode().getParentNode());
-            for (int c = 0; c < children.getLength(); c++){
-               QName childName = XmlUtils.getQName(children.item(c));
-               // Compare child QName to full part QName.
-               if (!fullPartName.equals(childName.toString())){
-                  XmlCursor cur = paths[0].newCursor();
-                  cur.toParent();
-                  cur.toChild( childName );
-                  errors.add( XmlError.forCursor( "Invalid element [" + childName + "] in SOAP Body", cur ) );
-                  cur.dispose();
-               }
-            }
-         }
-         log.debug("SoapBody validation errors: " + errors.size());
-      } 
-      return errors;
-   }
-   
-   /** Helper message for validating the body of a Soap message. */
-   private static void validateMessageBody(WsdlContext ctx, List<XmlError> errors, SchemaType type, XmlObject msg) throws XmlException {
-      // Need to create new body element of correct type from xml text since we want to retain line-numbers.
+  /** A commons logger for diagnostic messages. */
+  private static Logger log = LoggerFactory.getLogger(SoapMessageValidator.class);
```

Comment:
File was indented with 2 spaces instead of 3. Can you please fix this to avoid noise?